InnoGotchiBackend/InnnoGotchi.DAL/Respositories/PictureRepository.cs

```diff
@@ -0,0 +1,69 @@
+﻿using InnnoGotchi.DAL.EF;
+using InnnoGotchi.DAL.Entities;
+using InnnoGotchi.DAL.Interfaces;
+
+namespace InnnoGotchi.DAL.Respositories
+{
+    /// <summary>
+    /// Represents repository that has full access to picture entities in database
+    /// </summary>
+    public class PictureRepository : IRepository<Picture>
+    {
+        private InnoGotchiContext _context;
+
+        public PictureRepository(InnoGotchiContext context)
+        {
+            _context = context;
+        }
+        public bool Contains(Func<Picture, bool> predicate)
+        {
+            Picture? picture = FirstOrDefault(predicate);
+            if (picture == null)
+                return false;
+
+            return true;
+        }
+
+        public void Create(Picture item)
+        {
+            _context.Pictures.Add(item);
+        }
+
+        public bool Delete(int id)
+        {
+            Picture? picture = Get(id);
+            if (picture != null)
+            {
+                _context.Pictures.Remove(picture);
+                return true;
+            }
+            else
+                return false;
+        }
+
+        public IQueryable<Picture> FindAll(Func<Picture, bool> predicate)
+        {
+            return GetAll().Where(predicate).AsQueryable();
+        }
+
+        public Picture? FirstOrDefault(Func<Picture, bool> predicate)
```

Comment:
Why we need this wrapper?

Reply:
If you mean that wrapper is .AsQueryble() I already fixed that, else I don't understand what do you mean

Reply:
Why we need custom FirstOrDefault that call FirstOrDefault from LINQ?